mmv1/third_party/cai2hcl/common/utils.go

```diff
@@ -34,6 +35,17 @@ func DecodeJSON(data map[string]interface{}, v interface{}) error {
 	return nil
 }
 
+// MapToCtyValWithSchema normalizes and converts resource from untyped map format to TF JSON.
+//
+// Normalization is a post-processing of the output map, which does the following:
+// * Converts unmarshallable "schema.Set" to marshallable counterpart.
+// * Strips out properties, which are not part ofthe resource TF schema.
+func MapToCtyValWithSchemaNormalized(m map[string]interface{}, s map[string]*schema.Schema) (cty.Value, error) {
+	m = normalizeNodeRec(m, s).(map[string]interface{})
```

Comment:
Could we just add this to MapToCtyValWithSchema instead of making a separate function, since we want to always have this behavior in the long run? The error cases that this resolves are not intentional behavior - they are bugs that you're fixing.

Reply:
Done